src/core/src/pass/serialize.cpp

```diff
@@ -916,10 +916,21 @@ void serialize_rt_info(pugi::xml_node& root, const std::string& name, const ov::
         child.append_attribute("name").set_value(name.c_str());
     }
     if (data.is<std::shared_ptr<ov::Meta>>()) {
-        std::shared_ptr<ov::Meta> meta = data.as<std::shared_ptr<ov::Meta>>();
-        ov::AnyMap& map = *meta;
-        for (const auto& it : map) {
-            serialize_rt_info(child, it.first, it.second);
+        auto meta = data.as<std::shared_ptr<ov::Meta>>();
+        auto meta_with_pugixml_node = std::dynamic_pointer_cast<ov::MetaDataWithPugixml>(meta);
+        if (meta_with_pugixml_node) {
+            try {
+                auto pugi_node = meta_with_pugixml_node->get_pugi_node();
+                root.remove_child(child);
+                root.append_copy(pugi_node);
+            } catch (ov::Exception&) {
+                //Meta in MetaDataWithPugixml cannot be accessed by get_pugi_node. Read it as ov::AnyMap
```

Comment:
I suppose we need to go by the same code branch as `else` below?

Reply:
done